parser/enum.go

```diff
@@ -211,10 +196,12 @@ func (p *Parser) parseEnumBody() (
 				break
 			}
 
-			return nil, nil, scanner.Position{}, &parseEnumBodyStatementErr{
-				parseEnumFieldErr:      enumFieldErr,
-				parseEmptyStatementErr: emptyErr,
-			}
+			return nil, nil, scanner.Position{}, emptyErr
```

Comment:
> Certain internal errors get thrown on parse issues, which makes it impossible to open a switch statement on them.

I understand your need.

> This PR removes those, instead returning the latest error, which seems to be the most relevant from the parse tests I ran.
> As you can see, both errors are very similar. In my opinion, the second one is slightly more helpful; hence why i return that one. 

I thought this composite type was necessary to include enough errors so that end-users(or rather, maintainers) can see what input token is being expected.

In terms of usability, the error message should have looked like below.

```go
func (e *parseEnumBodyStatementErr) Error() string {
	return fmt.Sprintf(
		"expected enumField or emptyStatement and tried parsing each, "+
			"but following two errors cropped up: (1) parseEnumFieldErr`%v` (2) parseEmptyStatementErr`%v`",
		e.parseEnumFieldErr,
		e.parseEmptyStatementErr,
	)
}
```

>> expected enumField or emptyStatement and tried parsing each, but following two errors cropped up: (1) parseEnumFieldErr`found "\"option\"(Token=2, Pos=simple.proto:9:5)" but expected [=] at /GOPATH/src/github.com/yoheimuta/go-protoparser/parser/enum.go:242` (2) parseEmptyStatementErr`found "option" but expected [;]`

> makes it impossible to open a switch statement on them.

Is it not enough to prepare a method/interface for extracting inner errors?

```go
type CompositeError interface {
	Errors() []error
}
```

```go
func (e *parseEnumBodyStatementErr) Errors() []error {
	return []error{e.parseEnumFieldErr, e.parseEmptyStatementErr}
}
```